SourceX/dx.cpp

```diff
@@ -271,11 +272,20 @@ void NewHWCursor(SDL_Surface *surf)
 	if (renderer) {
 		// upscale
 		int ww, wh;
+		float ratio;
 
 		SDL_GetWindowSize(ghMainWnd, &ww, &wh);
 
-		scaled_w = scaled_w * ww / SCREEN_WIDTH;
-		scaled_h = scaled_h * ww / SCREEN_WIDTH;
+		float hFactor = float(wh) / SCREEN_HEIGHT;

```

Comment:
This looks a lot like the functions here: https://github.com/diasurgical/devilutionX/blob/52a8cd5fdb7cd8b74547f2850f474a20ab317124/SourceX/display.h#L43

Reply:
Erm, they don't very similar to the conversion functions in display.h to me. For one thing, those function deal with coordinates in the viewport, not with absolute dimensions.
The code does look quite similar to CalculatePreferdWindowSize but not quite the same.